tests/integration/netdev.bats

```diff
@@ -0,0 +1,262 @@
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	update_config ' .linux.netDevices |= {"dummy0": {} }'
+
+	runc run -d --console-socket "$CONSOLE_SOCKET" test_busybox
+	[ "$status" -eq 0 ]
+
+	# the network namespace owner controls the lifecycle of the interface
+	ip netns exec "$ns_name" ip link set dev dummy0 netns 1
+	[ "$status" -eq 0 ]
+
+	runc delete --force test_busybox
+
+	# verify the interface is back in the root network namespace
+	ip address show dev dummy0
+	[ "$status" -eq 0 ]
+}
+
+@test "move network device to precreated container network namespace" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+								| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+
+	# verify the interface is still present in the network namespace
+	ip netns exec "$ns_name" ip address show dev dummy0
+	[ "$status" -eq 0 ]
+
+	ip netns del "$ns_name"
+}
+
+@test "move network device to precreated container network namespace and set ip address" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+								| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	# set a custom address to the interface
+	# set the interface down to avoid network problems
+	ip link set down dev dummy0
+	ip address add 169.254.169.77 dev dummy0
+
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+	[[ "$output" == *"169.254.169.77"* ]]
+
+	# verify the interface is still present in the network namespace
+	ip netns exec "$ns_name" ip address show dev dummy0
+	[ "$status" -eq 0 ]
+	[[ "$output" == *"169.254.169.77"* ]]
+
+	ip netns del "$ns_name"
+}
+
+@test "move network device to precreated container network namespace and set ip address without global scope" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+								| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	# set a custom address to the interface
+	# set the interface down to avoid network problems
+	ip link set down dev dummy0
+	ip address add 127.0.0.33 dev dummy0
```

Comment:
What is the difference with the previous test?

Reply:
this is the IFA_A_PERMANENT test, 127.0.0.33 is not moved because is a loopback address